nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/AbstractSNMPProcessor.java

```diff
@@ -16,413 +16,249 @@
+                    " applies to get requests. The read-write community string applies to set requests. The trap" +
+                    " community string applies to receipt of traps.")
+            .required(true)
+            .sensitive(true)
             .defaultValue("public")
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V1, SNMP_V2C)
             .build();
 
-    /** property to define SNMP security level to use */
     public static final PropertyDescriptor SNMP_SECURITY_LEVEL = new PropertyDescriptor.Builder()
             .name("snmp-security-level")
             .displayName("SNMP Security Level")
-            .description("SNMP Security Level to use")
+            .description("SNMP version 3 provides extra security with User Based Security Model (USM). The three levels of security is " +
+                    "1. Communication without authentication and encryption (NoAuthNoPriv). " +
+                    "2. Communication with authentication and without encryption (AuthNoPriv). " +
+                    "3. Communication with authentication and encryption (AuthPriv). ")
             .required(true)
-            .allowableValues("noAuthNoPriv", "authNoPriv", "authPriv")
-            .defaultValue("authPriv")
+            .allowableValues(NO_AUTH_NO_PRIV, AUTH_NO_PRIV, AUTH_PRIV)
+            .defaultValue(NO_AUTH_NO_PRIV.getValue())
+            .dependsOn(SNMP_VERSION, SNMP_V3)
             .build();
 
-    /** property to define SNMP security name to use */
     public static final PropertyDescriptor SNMP_SECURITY_NAME = new PropertyDescriptor.Builder()
             .name("snmp-security-name")
-            .displayName("SNMP Security name / user name")
-            .description("Security name used for SNMP exchanges")
-            .required(false)
+            .displayName("SNMP Security Name / Username for SNMP v3 Authentication")
+            .description("Security name / User name used for SNMP exchanges.")
+            .required(true)
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V3)
             .build();
 
-    /** property to define SNMP authentication protocol to use */
-    public static final PropertyDescriptor SNMP_AUTH_PROTOCOL = new PropertyDescriptor.Builder()
-            .name("snmp-authentication-protocol")
-            .displayName("SNMP Authentication Protocol")
-            .description("SNMP Authentication Protocol to use")
+    public static final PropertyDescriptor SNMP_PRIVACY_PROTOCOL = new PropertyDescriptor.Builder()
+            .name("snmp-private-protocol")
+            .displayName("SNMP Privacy Protocol")
+            .description("Privacy allows for encryption of SNMP v3 messages to ensure confidentiality of data.")
             .required(true)
-            .allowableValues("MD5", "SHA", "")
+            .allowableValues(DES, DES3, AES128, AES192, AES256, NO_ENCRYPTION)
```

Comment:
`None` option seems to be obsolete because this property depends on Security Level and when the property is visible, it is mandatory.